ibis/backends/flink/__init__.py

```diff
@@ -303,7 +304,16 @@ def get_schema(
         qualified_name = sg.table(table_name, db=catalog, catalog=database).sql(
             self.name
         )
-        table = self._table_env.from_path(qualified_name)
+        try:
+            table = self._table_env.from_path(qualified_name)
+        except Py4JJavaError as e:
+            # This seems to msg specific but not sure what a good work around is
+            if re.search(
+                rf"Table `{re.escape(table_name)}` was not found",
```

Comment:
I don't like this, but I'm not sure if there is a better way to get the error matching here. Suggestions welcomed.

Reply:
@cpcloud do we think this is alright for flink? It's a bit specific but I didn't find an alternative, If so I'll remove the comment here

Reply:
It seems like there's some poking around the java object that could be done.

Reply:
I tried multiple thinks, I wasn't able to land in anything better than this, I tried inspecting `e.java_exception` to get the `getSQLState` but I wasn't able. Maybe I'm missing something.

Reply:
I went ahead and changed this to catch the error, and then check for table existence instead of matching.

`INVALID_INPUT` is way too general of a string to match on in any case.